File: arb-www/src/schema/mlb/standings.ts
import { BaseDTO } from '../BaseDTO';

/**
 * MLB Standings information for a single team
 */
export class MLBStanding extends BaseDTO {
  Season!: number;
  SeasonType!: number;
  TeamID!: number;
  Key!: string;
  City!: string | null;
  Name!: string;
  League!: string;
  Division!: string;
  Wins!: number;
  Losses!: number;
  Percentage!: number;
  DivisionWins!: number;
  DivisionLosses!: number;
  GamesBehind!: number | null;
  LastTenGamesWins!: number;
  LastTenGamesLosses!: number;
  Streak!: string | null;
  LeagueRank!: number;
  DivisionRank!: number;
  WildCardRank!: number | null;
  WildCardGamesBehind!: number | null;
  HomeWins!: number;
  HomeLosses!: number;
  AwayWins!: number;
  AwayLosses!: number;
  DayWins!: number | null;
  DayLosses!: number | null;
  NightWins!: number | null;
  NightLosses!: number | null;
  RunsScored!: number;
  RunsAgainst!: number;
  GlobalTeamID!: number;
}

/**
 * MLB Standings response
 */
export class MLBStandingsResponse extends BaseDTO {
  data!: MLBStanding[];
}

File: arb-rs/src/schema/sportradar/mlb/standings.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, Deserialize, Clone)]
#[serde(rename_all = "PascalCase")]
pub struct MLBStanding {
    pub season: i32,
    pub season_type: i32,
    #[serde(rename = "TeamID")]
    pub team_id: i32,
    pub key: String,
    pub city: Option<String>,
    pub name: String,
    pub league: String,
    pub division: String,
    pub wins: i32,
    pub losses: i32,
    pub percentage: f64,
    pub division_wins: i32,
    pub division_losses: i32,
    pub games_behind: Option<f64>,
    pub last_ten_games_wins: i32,
    pub last_ten_games_losses: i32,
    pub streak: Option<String>,
    pub league_rank: i32,
    pub division_rank: i32,
    pub wild_card_rank: Option<i32>,
    pub wild_card_games_behind: Option<f64>,
    pub home_wins: i32,
    pub home_losses: i32,
    pub away_wins: i32,
    pub away_losses: i32,
    pub day_wins: Option<i32>,
    pub day_losses: Option<i32>,
    pub night_wins: Option<i32>,
    pub night_losses: Option<i32>,
    pub runs_scored: i32,
    pub runs_against: i32,
    #[serde(rename = "GlobalTeamID")]
    pub global_team_id: i32,
}

pub type MLBStandings = Vec<MLBStanding>;
